tests/functional/eks/test_get_token.py

```diff
@@ -143,3 +144,105 @@ def test_url_different_partition(self):
             expected_endpoint='sts.cn-north-1.amazonaws.com.cn',
             expected_signing_region='cn-north-1'
         )
+
+    def test_api_version_discovery_deprecated(self):
+        os.environ["KUBERNETES_EXEC_INFO"] = '{"kind":"ExecCredential","apiVersion":"client.authentication.k8s.io/v1alpha1","spec":{"interactive":true}}'
```

Comment:
Instead of mutating the global environment variables and cleaning them up after the assertions for each case, this test class already patches the `os.environ`: https://github.com/aws/aws-cli/blob/7a1da123217518faa1d850f23d0f819db2a04c84/awscli/testutils.py#L323-L332

So for all of these tests, we should be able to set the environment variable as follows:
```python
self.environ['KUBERNETES_EXEC_INFO'] = ...  # JSON document
```
This allows us to set the environment variables without risk of mutating the global state and also removes the need to delete it from the global env vars at the end.